src/Compilers/Core/Portable/DiagnosticAnalyzer/IAnalyzerAssemblyRedirector.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.Diagnostics.Redirecting;
```

Comment:
Why did we put this into a separate namespace?

Reply:
So we can use RestrictedInternalsVisibleTo to make this (and only this) namespace visible to the SDK counterpart.